### PR TITLE
[MPS] Add huber loss

### PR DESCRIPTION
**Pull request by qqaatw**
Fixes #ISSUE_NUMBER

### CODE REVIEW

File: aten/src/ATen/native/mps/operations/LossOps.mm [hunk 1]
```diff
@@ -1035,6 +1035,242 @@ void smooth_l1_loss_backward_template(
...
+                                                                                       name:nil];
+                    MPSGraphTensor* outputTensor = [mpsGraph selectWithPredicateTensor: [mpsGraph lessThanOrEqualToWithPrimaryTensor: diffTensor
+                                                                                                                     secondaryTensor:[mpsGraph negativeWithTensor: deltaTensor
+                                                                                                                                                          name: nil]
+                                                                                                                                name:nil]
+                                                                   truePredicateTensor: firstCondTensor
+                                                                  falsePredicateTensor: secondThirdTensor
+                                                                                  name:nil];
+
+                    //outputTensor = normGradOutputDeltaTensor;
```
Comment (kulinseth) [inline aten/src/ATen/native/mps/operations/LossOps.mm:1245]: Cleanup.

**Review by kulinseth — APPROVED** [review]
Overall looks good, please cleanup. And add the formula or a link for the formula for the backward pass of Huber loss.

**Comment by qqaatw** [comment]
@albanD should we add `ciflow/trunk` label?

**Comment by kulinseth** [comment]
> @albanD should we add `ciflow/trunk` label?

Done.

**Comment by albanD** [comment]
@qqaatw we should always have this label yes as the MPS test only run on trunk and we want them to run on MPS PRs. Don't hesitate to remind us to add it if we forget.

The newly added test fail in CI btw.

**Comment by qqaatw** [comment]
@albanD OK! The failure was due to a typo lol.

File: test/test_mps.py [hunk 2]
```diff
@@ -1799,6 +1799,58 @@ def helper(shape, reduction):
         helper([7, 5, 2, 4, 6], 'sum')
         helper([8, 4, 5, 7, 6], 'mean')
 
+    # Huber Loss
+    def test_huber_loss(self):
```
[the remark this answers is not in the document]
Comment (qqaatw) [reply test/test_mps.py]: Sounds good

**Review by malfet — APPROVED** [review]

**Pull request closed without merging**